implementations/rust/ockam/ockam_api/src/nodes/service/message.rs

```diff
@@ -44,21 +43,20 @@ impl NodeManagerWorker {
         &self,
         ctx: &Context,
         req: &RequestHeader,
-        dec: &mut Decoder<'_>,
-    ) -> Result<Vec<u8>> {
-        let req_body: SendMessage = dec.decode()?;
-        let multiaddr = req_body.multiaddr()?;
-        let msg = req_body.message.to_vec();
+        send_message: SendMessage,
+    ) -> Result<Response<Vec<u8>>, Response<Error>> {
+        let multiaddr = send_message.multiaddr()?;
+        let msg = send_message.message.to_vec();
 
         let res = self
             .node_manager
             .send_message(ctx, &multiaddr, msg, None)
             .await;
         match res {
-            Ok(r) => Ok(Response::ok(req).body(r).to_vec()?),
+            Ok(r) => Ok(Response::ok(req).body(r)),
             Err(err) => {
                 error!(target: TARGET, ?err, "Failed to send message");
-                Ok(Response::internal_error(req, "Failed to send message").to_vec()?)
+                Err(Response::internal_error(req, "Failed to send message"))
```

Comment:
> I'm finding very confusing the fact that we can return an Ok(()), an Ok(Response::some_error()) and an Err(Response::some_err()).

I don't understand where exactly we do this. In general the `NodeManagerWorker` results have the form `Result<Response<T>, Response<Error>>`. Where are the cases with `Ok(Response::error(...))`? (if you also account for the 2 other PRs which are aligning the rest of the code).

Reply:
I'm talking about things that have been there for a long time, it's not anything particular from these PRs, but I'm pointing this out in case now is a good time to reconsider how we handle certain responses.

This is one example: 

https://github.com/build-trust/ockam/blob/f6ef683b51140eb975718edb02970800cf34f766/implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor.rs#L41

Reply:
Ah, yes! It would be nice indeed to stop returning `Vec<u8>` everywhere :-)